### PR TITLE
feat: Added Individual item Product page

### PR DESCRIPTION
**Pull request by Naman-Saxena1**
# Added individual product page for each item
Now when user clicks on item card on Home or Shop Page, its Product page opens on new tab ( just like Amazon, Flipkart and Myntra ).

# Product Page Examples
Capture1
Capture2

# Files to review :
1. App.js
2. ProductCard.js
4. ProductPage.css
3. ProductPage.jsx

### CODE REVIEW

File: src/Components/Card/ProductCard.js [hunk 1]
```diff
@@ -1,36 +1,51 @@
...
-                <div className="item-title">
-                    <h4>{bookname}</h4>
+        <Link 
+            to={`/shop/${id}`}  
+            onClick={() => localStorage.setItem(`${id}`, JSON.stringify(productdetails))}
+            target="_blank"
+            rel="noopener noreferrer"
+        >
+            <div className="card-basic">
+                <img src={imgsrc} alt={imgalt}/>
```
Comment (gauravsinhaweb) [inline src/Components/Card/ProductCard.js]: using props in the capital alphabet can give more readability.

File: src/Pages/ProductPage/ProductPage.jsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        author,
+        originalprice,
+        discountedprice,
+        discountpercent,
+        imgsrc, 
+        imgalt,
+        badgetext, 
+        outOfStock,
+        rating, 
+        description
```
Comment (gauravsinhaweb) [inline src/Pages/ProductPage/ProductPage.jsx]: either use camelCase or capital alphabets text styling for all the props for readibility.

File: src/Components/Card/ProductCard.js [hunk 3]
```diff
@@ -1,36 +1,51 @@
 import React from 'react'
+import { Link } from "react-router-dom"
 
 export default function ProductCard(props) 
 {
     let {productdetails} = props;
 
-    let {bookname,author,originalprice,discountedprice,discountpercent,imgsrc, imgalt,badgetext, outOfStock} = productdetails
+    let {id, bookname,author,originalprice,discountedprice,discountpercent,imgsrc, imgalt,badgetext, outOfStock} = productdetails
```
Comment (bhtibrewal) [inline src/Components/Card/ProductCard.js]: use const for this you are not going to change them

**Review by gauravsinhaweb — APPROVED** [review]
Good work!, UI looks so clean, just added some minor points for changes. Good to merge after changes!

**Review by bhtibrewal — APPROVED** [review]
good work Naman, just one mistake correct it then good to merge

File: src/Components/Card/ProductCard.js (hunk 1 above)
Comment (Naman-Saxena1) [reply src/Components/Card/ProductCard.js]: Done.

File: src/Pages/ProductPage/ProductPage.jsx (hunk 2 above)
Comment (Naman-Saxena1) [reply src/Pages/ProductPage/ProductPage.jsx]: Done.

File: src/Components/Card/ProductCard.js (hunk 3 above)
Comment (Naman-Saxena1) [reply src/Components/Card/ProductCard.js]: Done.

**Pull request merged**